src/sessions.ts

```diff
@@ -447,6 +447,15 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    * Checkout a descriptive example here:
    * @see https://www.mongodb.com/blog/post/quick-start-nodejs--mongodb--how-to-implement-transactions
    *
+   * If a command inside this function fails:
+   * - It may cause the transaction on the server to be aborted.
+   * - This situation is normally handled transparently by the driver.
+   * - However, if the application catches such an error and does not re-raise it, the driver will not be able to determine whether the transaction was aborted or not.
+   * - The driver will then retry the transaction indefinitely.
+   *
+   * To avoid this situation, the application must not silently handle errors within this function.
+   * If the application needs to handle errors within, it must await all operations such that if an operation is rejected it becomes the rejection of the callback function passed into withTransaction.
```

Comment:
Not on this line but above, I want the "quick" important sentence to call out this detail too if that sounds good. I think you can leave what is written here as is.

> **IMPORTANT:** This method requires the user to return a Promise, and `await` all operations.

update to:

> **IMPORTANT:** This method requires the function passed in to return a Promise. That promise must be made by `await`-ing all operations in such a way that rejections are propagated to the returned promise.